package.json

```diff
@@ -20,24 +23,36 @@
     "mocha": "^4.0.1",
     "morgan": "^1.9.0",
     "node-sass": "^4.7.2",
+    "parallelshell": "^3.0.2",
     "promise-map-series": "^0.2.3",
     "serve-favicon": "^2.4.5",
-    "supertest": "^3.0.0"
+    "supertest": "^3.0.0",
+    "watch": "^1.0.2",
+    "webpack": "^3.8.1",
+    "webpack-merge": "^4.1.1"
   },
   "scripts": {
     "clean": "rm -rf dist",
     "lint": "./node_modules/.bin/eslint src/**/*.js",
     "build:static:styles": "./scripts/build-static-styles.sh",
-    "build:static:scripts": "mkdir -p dist/static/js && cp -R src/client/js/* dist/static/js",
-    "build:static": "npm run build:static:styles && npm run build:static:scripts",
+    "watch:static:styles": "./node_modules/.bin/watch 'npm run build:static:styles' src/client/styles",
+    "build:static:scripts": "./scripts/build-static-scripts.sh",
+    "watch:static:scripts": "./node_modules/.bin/watch 'npm run build:static:scripts' src/client/js src/shared/js",
+    "build:static": "./node_modules/.bin/parallelshell 'npm run build:static:styles' 'npm run build:static:scripts'",
+    "watch:static": "./node_modules/.bin/parallelshell 'npm run watch:static:styles' 'npm run watch:static:scripts'",
     "build:node:scripts": "./scripts/build-node-scripts.sh",
-    "build:node:views": "mkdir -p dist/node/shared/views && cp -R src/shared/views/* dist/node/shared/views",
-    "build:node": "npm run build:node:scripts && npm run build:node:views",
-    "build": "npm run clean && npm run build:static && npm run build:node",
-    "serve": "npm run build && node scripts/serve.js",
-    "test:js": "./node_modules/.bin/karma start",
+    "watch:node:scripts": "./node_modules/.bin/watch 'npm run build:node:scripts' src/server/js src/shared/js data",
+    "build:node:views": "mkdir -p dist/node/shared/views && mkdir -p dist/node/shared/partials && cp -R src/shared/views/* dist/node/shared/views && cp -R src/shared/partials/* dist/node/shared/partials",
+    "watch:node:views": "./node_modules/.bin/watch 'npm run build:node:views' src/shared/views src/shared/partials",
+    "build:node": "./node_modules/.bin/parallelshell 'npm run build:node:scripts' 'npm run build:node:views'",
+    "watch:node": "./node_modules/.bin/parallelshell 'npm run watch:node:scripts' 'npm run watch:node:views'",
+    "build": "npm run clean && ./node_modules/.bin/parallelshell 'npm run build:static' 'npm run build:node'",
+    "build:prod": "NODE_ENV=production npm run build",
+    "watch": "./node_modules/.bin/parallelshell 'npm run watch:static' 'npm run watch:node'",
+    "serve": "./node_modules/.bin/parallelshell 'npm run watch' 'node scripts/serve.js'",
     "test:node": "./scripts/test-node.sh",
-    "test": "npm run test:node"
+    "test:static": "./node_modules/.bin/karma start",
+    "test": "npm run test:node && npm run test:static"
```

Comment:
Sorry about that, should have checked that before creating a PR. Looks like this is due to the directory structure changes, since we end up with slightly different paths inside `dist`.

Should we move `data` inside `src` to avoid this problem? It probably makes sense anyway, since we're directly importing it into other modules, right?